transmon_pocket_teeth.py

```diff
@@ -0,0 +1,330 @@
+                              _qgeometry_table_path='True',
+                              _qgeometry_table_poly='True',
+                              _qgeometry_table_junction='True')
+    """Component metadata"""
+
+    TOOLTIP = """Transmon pocket with 6 connection pads."""
+
+    def make(self):
+        """Define the way the options are turned into QGeometry.
+
+        The make function implements the logic that creates the geoemtry
+        (poly, path, etc.) from the qcomponent.options dictionary of
+        parameters, and the adds them to the design, using
+        qcomponent.add_qgeometry(...), adding in extra needed
+        information, such as layer, subtract, etc.
+        """
+        self.make_pocket()
+        self.make_connection_pads()
+
+    def make_pocket(self):
+        """Makes standard transmon in a pocket."""
+
+        # self.p allows us to directly access parsed values (string -> numbers) form the user option
+        p = self.p
+      #  pcop = self.p.coupled_pads[name]  # parser on connector options
+
+        # since we will reuse these options, parse them once and define them as variables
+        pad_width = p.pad_width
+        pad_height = p.pad_height
+        pad_gap = p.pad_gap
+        coupled_pad_height = p.coupled_pad_height 
+        coupled_pad_width = p.coupled_pad_width 
+        coupled_pad_gap = p.coupled_pad_gap
+
+                                 
+
+        # make the pads as rectangles (shapely polygons)
+        pad = draw.rectangle(pad_width, pad_height)
+              
+        pad_top = draw.translate(pad, 0, +(pad_height + pad_gap) / 2.)
+        # Here, you make your pads round. Not sharp shape on the left and right sides and also this should be the same for the bottom pad as the top pad.
+        circ_left_top = draw.Point(-pad_width/2., +(pad_height + pad_gap) / 2.).buffer(pad_height/2, resolution=16, cap_style=CAP_STYLE.round)
+        circ_right_top = draw.Point(pad_width/2., +(pad_height + pad_gap) / 2.).buffer(pad_height/2, resolution=16, cap_style=CAP_STYLE.round)
+        # In here you create the teeth part and then you union them as one with the pad. Teeth only belong to top pad. 
+        coupled_pad = draw.rectangle(coupled_pad_width, coupled_pad_height+pad_height)
+        coupler_pad_round = draw.Point(0., (coupled_pad_height+pad_height)/2).buffer(coupled_pad_width/2, resolution=16, cap_style=CAP_STYLE.round)
+        coupled_pad = draw.union(coupled_pad, coupler_pad_round)
+        coupled_pad_left = draw.translate(coupled_pad, -(coupled_pad_width/2. + coupled_pad_gap/2.), +coupled_pad_height/2. +pad_height + pad_gap/2.-pad_height/2 )
+        coupled_pad_right = draw.translate(coupled_pad, (coupled_pad_width/2. + coupled_pad_gap/2.), +coupled_pad_height/2. +pad_height + pad_gap/2.-pad_height/2 )
+        pad_top_tmp = draw.union([circ_left_top, pad_top, circ_right_top])
+        # The coupler pads are only created if low_W=0 and low_H=+1

```

Comment:
loc_W I presume?

Reply:
Yeah, because the design has the teeth part only in the middle of the top pad.